### DISCUSSION TITLE
sql functions and privileges?

### DISCUSSION THREAD

**Discussion by pierroo** [supabase/supabase#20521] [Questions] [answered]
I have created a function from a sql query.
I call/invoke it from my front (RN), with an authenticated user
I have RLS in place, that allows select on my own user only (based on jwt sub uuid)

HOWEVER, for the function invoked here, I would like it to bypass RLS
How can I do that?

Since I want this function to be able to read other users info (but I don't want the user to do it from the front for example, thus the RLS)
and run other stuff that wouldn't be allowed by RLS otherwise

Edit: I read somewhere that I can choose which privileges the function should run with (the invoker or the definer (meaning admin)); but I don't see how? and if that's actually the way to go?

**Comment by GaryAustin1** [comment, marked as answer]
Yes you use a security definer type function.  It is a setting in the function UI for new functions and a function parameter in SQL.  
https://supabase.com/docs/guides/database/functions#security-definer-vs-invoker

**Reply by pierroo** [reply]
Works perfectly, once again thank you so much for your fast reply!
